### PR TITLE
feat: breaking out the examples

### PR DESCRIPTION
**Pull request by subirjolly**
* What this PR does
This PR is meant to componentize client library code/examples so that we can take our flux queries and give our customers the option to view or copy the client library code with our flux. Instead of having hard-coded examples, we want to convert existing code to be dynamically generated so that we can reuse our code to display examples as well as the "Copy to Clipboard" client library code using our flux.

* Why this PR is necessary
To maintain consistency between our examples and "Copy to Clipboard" feature for Client Libraries, removing hard-coded examples is needed so that those 2 areas don't go out of sync and we won't have to update same code in 2 different places.

* What this PR affects
All client library examples which were hard-coded in MarkDown files. As far as the UI is concerned, there shouldn't be any different to the users.

* Potentially an image / gif of this in action
N/A

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1333)

**Comment by Asalem1** [comment]
Is this a draft PR?

**Comment by subirjolly** [comment]
> Is this a draft PR?

@drdelambre worked on it. I just rebased it and am trying to merge it to continue working on #576 
This PR has some base components/pre-reqs which are needed for the story. @drdelambre Feel free to elaborate or add.

**Review by Asalem1 — CHANGES_REQUESTED** [review]
This PR touched 118 files and has a lot of code deleted / modified. I'd love a PR description to give me a better idea of:

- What this PR does
- Why this PR is necessary
- What this PR affects
- Potentially an image / gif of this in action

**Review by Asalem1 — COMMENTED** [review]
Would love an image to validate that everything still works, maybe some e2e tests

File: src/writeData/clients/Kotlin/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import logo from 'src/writeData/clients/Kotlin/logo.svg'
+import description from 'src/writeData/clients/Kotlin/description.md'
+import initialize from 'src/writeData/clients/Kotlin/initialize.example'
+import execute from 'src/writeData/clients/Kotlin/execute.example'
+
+export default register =>
+  register({
+    id: 'kotlin',
+    name: 'Kotlin',
+    featureFlag: 'client-library--kotlin',
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/writeData/clients/Kotlin/index.ts:10]: @subirjolly was this one of your conflicts?

File: src/shared/containers/index.tsx [hunk 2]
```diff
@@ -58,15 +67,6 @@ export const MembersIndex = lazy(() =>
 export const RouteToDashboardList = lazy(() =>
   import('src/dashboards/components/RouteToDashboardList')
 )
-export const ClientLibrariesPage = lazy(() =>
-  import('src/writeData/containers/ClientLibrariesPage')
-)
-export const FileUploadsPage = lazy(() =>
-  import('src/writeData/containers/FileUploadsPage')
-)
-export const TelegrafPluginsPage = lazy(() =>
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/shared/containers/index.tsx:67]: yes

[the remark this answers is not in the document]
Comment (Asalem1) [reply src/shared/containers/index.tsx:67]: it looks like the imports were just moved from one part of the file to another.

**Comment by drdelambre** [comment]
re feature flags: wut?
re line protocol: wut?

File: src/writeData/clients/Kotlin/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (subirjolly) [reply src/writeData/clients/Kotlin/index.ts:10]: Don't think so. It was just the markdowns removal

**Comment by subirjolly** [comment]
> This PR touched 118 files and has a lot of code deleted / modified. I'd love a PR description to give me a better idea of:
> 
> * What this PR does
> * Why this PR is necessary
> * What this PR affects
> * Potentially an image / gif of this in action

Absolutely. Just prepared a draft. Getting it taken a look by Boaty.

**Comment by Asalem1** [comment]
> re feature flags: wut?
> re line protocol: wut?

Each of the files has a unique feature flag associated with it. Is there a reason we don't wanna lump em under section-specific ones?

Re: line protocol, all the sections we're adding have a blurb about writing data with line protocol. Is the client library section about integrating a client library? Why are we giving the user an ability to write line protocol? Maybe I'm misunderstanding things a bit here

**Comment by subirjolly** [comment]
> This PR touched 118 files and has a lot of code deleted / modified. I'd love a PR description to give me a better idea of:
> 
> * What this PR does
> * Why this PR is necessary
> * What this PR affects
> * Potentially an image / gif of this in action

done!

File: src/writeData/clients/Kotlin/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (subirjolly) [reply src/writeData/clients/Kotlin/index.ts:10]: This file is not in master.

**Comment by drdelambre** [comment]
re line protocol: i have no idea what is going on so lemme rewind
for all of the client libraries, we have one markdown file to teach the user how to use said library that has a description, information on initializing the library to talk to the server, 0-n ways on how to write to the db using the client, and an example of how to use the client library to execute a flux query.
this pr extracts those examples and puts them into `.example` files so that we can reuse them later on other pages
i have no opinion about whether or not we should provide the user with an example on how to use line protocol, just that if we provided it before decomposing the markdown file, we should still provide it after.

**Comment by drdelambre** [comment]
re feature flags: the reason it switched perspective from section level to client library level is that by introducing the concept of reusing these examples, we're implicitly declaring the ability to create side effects external to the existing implementation / visualization of these libraries. if you want to turn off the swift client examples (for... example), you'd want to turn them off not only for this page, but for every other page that implements them (notebooks `export as swift` being an upcoming one).

File: src/shared/containers/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/shared/containers/index.tsx:67]: yes

File: src/writeData/clients/Kotlin/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/writeData/clients/Kotlin/index.ts:10]: https://github.com/influxdata/ui/blob/master/src/writeData/components/clientLibraries/Kotlin.md

**Review by Asalem1 — APPROVED** [review]

**Comment by subirjolly** [comment]
I also clicked around in most of the application and everything seems to be normal.

**Pull request merged**